plugins/modules/identity/keycloak/keycloak_clientsecret.py

```diff
@@ -0,0 +1,236 @@
+              providing it will reduce the number of API calls required.
+        type: str
+    client_id:
+        description:
+            - The client_id of the client to lookup account client ID
+        aliases:
+          - clientId
+        type: str
+
+
+extends_documentation_fragment:
+- community.general.keycloak
+
+author:
+    - Fynn Chen (@fynncfchen)
+'''
+
+EXAMPLES = '''
+- name: Get a Keycloak client secret, authentication with credentials
+  community.general.keycloak_clientsecret:
+    id: '9d59aa76-2755-48c6-b1af-beb70a82c3cd'
+    realm: MyCustomRealm
+    state: present
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+  delegate_to: localhost
+
+- name: Generate a new Keycloak client secret, authentication with token
+  community.general.keycloak_user:
+    id: '9d59aa76-2755-48c6-b1af-beb70a82c3cd'
+    realm: MyCustomRealm
+    state: absent
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    token: TOKEN
+  delegate_to: localhost
+'''
+
+RETURN = '''
+msg:
+    description: Message as to what action was taken.
+    returned: always
+    type: str
+
+end_state:
+    description: Representation of the client credential after module execution (sample is truncated).
+    returned: on success
+    type: complex
+    contains:
+        type:
+            description: Credential type.
+            type: str
+            returned: always
+            sample: secret
+        value:
+            description: Secret of the client.
+            type: str
+            returned: always
+            sample: cUGnX1EIeTtPPAkcyGMv0ncyqDPu68P1
+
+clientsecret:
+  description:
+      - Representation of the client credential after module execution.
+      - Deprecated return value, it will be removed in community.general 6.0.0. Please use the return value I(end_state) instead.
+  returned: always
+  type: complex
+  contains:
+      type:
+          description: Credential type.
+          type: str
+          returned: always
+          sample: secret
+      value:
+          description: Secret of the client.
+          type: str
+          returned: always
+          sample: cUGnX1EIeTtPPAkcyGMv0ncyqDPu68P1
+
+'''
+
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import KeycloakAPI, camel, \
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        state=dict(default='present', choices=['present', 'absent']),
+        realm=dict(default='master'),
+
+        id=dict(type='str'),
+        client_id=dict(type='str', aliases=['clientId']),
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
```

Comment:
From the examples, it *looks like* one must pass `(auth_realm, auth_username, auth_password)` or `token`, but not both.

If that's indeed the case, there should be a `mutually_exclusive` clause here.